### DISCUSSION TITLE
New Resource Manager observations

### DISCUSSION THREAD

**Discussion by vmasanas** [dnnsoftware/Dnn.Platform#4214] [Ideas]
Hi,
I like the new resource manager module but I've found a couple things that I think would improve a lot the experience.

1. As far as I can see there's only a single view with the big icons. I'd suggest adding an alternate view showing all details (name, date created, date modified, size, ...). And then allow users to alternate between list view and icon view

2. There's a drop down with values Item name, Last modified on date, created on date. 
I guess this is the sorting of items but notice it does not say it anywhere. 
Suggestion would be to change the text to "Order by..." 
It would also be nice to have an ASC/DESC option
It would also be nice to have a sort by size option

3. User of popups not consistent
I see a few options that open a popup, for example when trying to delete an item.
Any other option seems to open a new section at the end of the listing. I find it quite confusing, even more if the list of items in the folder is large. It's not obvious that you click on edit penzil for example and then you need to scroll down to see the details of the item.
My suggestion would be to open all those "editor" sections in a popup.

Also notice that if you click the edit penzil on the first item of a two row list of items, the second rows gets moved.
This would be fixed if we used popups instead, so not a big deal if you choose to change to popups.

![imagen](https://user-images.githubusercontent.com/4357679/96365276-bc9dc480-113f-11eb-8025-c4d0b7544cce.png)

4. Multiple seleccion
This is related to number 1 above.
There's no option to select multiple items at once. So if you want to move, delete, download, ... you need do to that one by one.
I'd suggest a selectbox on the list view to allow for this option.

5. Edit and move should close previous edit sections
Not an issue if we used popups as discussed in number 3
Take a file, and click the penzil icon to edit. Then click the icon to move.
You end up with two sections option at the bottom of the page.
Very confusing

6. Permissions for files
When editing a folder I see the option to manage permissions.
That doesn't seem to be available when editing a file.

7. 
Create a folder TEST and add a file to it.
Click edit iconon the file. Without closing the edit section below click the parent folder on the breadcrumb
Click move icon on the folder.
You end up with two open sections at the botton: one to edit the previous file, one to move the folder.
Again, this all would be solved if using modal popups for editors.

8. Spacing
Some spacing required here
![imagen](https://user-images.githubusercontent.com/4357679/96365505-544fe280-1141-11eb-92a0-ae4d24b399ae.png)

**Comment by valadas** [comment]
Hi @vmasanas 

This issue is a bit broad with many different things. most of the reported UI issues we have already discussed and are planning to improve, but it was a bit of a massive undertaking to get this to about equal feature parity for the most important items. Much of what you mention we have already plans to improve, it will just not be in the very next release unless it's a major issue.

So, first, for file permissions, I don't think this was an option in the Digital Assets Manager or even a supported scenario in Dnn, are you talking in the context of secure folders ?

Second, if you don't mind splitting these multiple things into smaller issues, I am sure it would help a lot in developers picking up as an issue this size might be quite a large undertaking as it is.

**Comment by vmasanas** [comment]
@valadas I understand that you might not be able to address all items at once, but I just wanted to be sure we at least have the conversation open and considered. I think my points are quite understandable, and I'm happy to see you say you already considered some of them.

On the permission I see you are right: this was not available on the previous module either. 

On splitting the topic into multiple tickets.... notice that most of it resolves to implementing popups so maybe I can group a few into a single ticket. Notice this was not really an issue report ticket but a way to get some conversation on the topics I saw.

**Comment by david-poindexter** [comment]
Thank you @vmasanas for the confirmation on a lot we have already discussed and have similar thoughts on.  It is great to get it in writing somewhere.  Perhaps we convert this "issue" into a "discussion".  Then smaller, actionable "issues" can be created as a follow on.

**Comment by thienvc** [comment]
I also want the same.  It's great that the new File Manager is much faster, but UI and UX are still as bad as DNN's style ever (PersonalBar UI, Mobile mode UI, ...).  Using DNN on mobile is a pain.  It seems that DNN FrontEnd UI does not have any common standards at all 😅😅

**Comment by sleupold** [comment]
I agree, permission per file would fit better with windows permission, but this hasn't been implemented in DNN Platform library, as there was little request and managing permissions can become quite cumbersome, as long as we don't support inheritance (which would be the next can of worms and a huge undertake). Just my 2 cents.
